### DISCUSSION TITLE
bad fitting of force when force is large in a MLP for ionized water

### DISCUSSION THREAD

**Discussion by beidouamg** [deepmodeling/deepmd-kit#3265] [Q&A]
Hello DeePMDer,

I am a DeePMD beginner and trying to train a potential capable of capturing the dissociation of water. 
To build my training set, I use metadynamics to get the structures contain OH- and H3O+ in the water solution. I'm sure that my DFT calculations are converged and accurate, the whole cell is neutral, the total number of structures for training are more than 50000 and they are as diverse as possible. 
After training, I got the results as below, the fitting for energy is somehow good, **but the fitting for force deviates from the ideal values, especially when the forces are large**.

I am not sure if DeePMD can capture the feature of charged ions. I would appreciate if you can point out the possible problem in my training. The input for DeepMD training is pasted below:

{
    "model": {
        "type_map":     ["H", "O"],
        "descriptor" :{
            "type":             "se_e2_a",
            "sel":              "auto",
            "rcut_smth":        0.50,
            "rcut":             6.50,
            "neuron":           [30, 60, 120],
            "resnet_dt":        false,
            "axis_neuron":      16,
            "seed":             11300402      },
        "fitting_net" : {
            "type":             "ener",
            "neuron":           [240, 240, 240, 240],
            "resnet_dt":        true,
            "seed":             11300402      }},

    "learning_rate" :{
        "type":         "exp",
        "decay_steps":  15000,
        "start_lr":     0.001,
        "stop_lr":      3.51e-8  },

    "loss" :{
        "type"        :"ener",
        "start_pref_e": 0.02,
        "limit_pref_e": 1000,
        "start_pref_f": 1000,
        "limit_pref_f": 1000,
        "start_pref_v": 0,
        "limit_pref_v": 0   },

    "training" : {
        "training_data": {
            "systems":     ["../training_data/230K10",
                            "../training_data/330K10",
                            "../training_data/330K11",
                            "../training_data/430K10",
                            "../training_data/530K10"],
            "batch_size":  "auto",
            "auto_prob":   "prob_sys_size" },
        "validation_data":{
            "systems":     ["../validation_data/230K10",
                            "../validation_data/330K10",
                            "../validation_data/330K11",
                            "../validation_data/430K10",
                            "../validation_data/530K10"],
            "batch_size":  "auto",
            "numb_btch":   1     },
        "numb_steps":    3000000,
        "disp_file":    "lcurve.out",
        "disp_freq":     100,
        "save_freq":     1000,
        "seed":          11300402      }
}
![energy](https://github.com/deepmodeling/deepmd-kit/assets/51835970/8b6ac84c-c6fa-4aac-861f-fd5e9e3994e6)
![force](https://github.com/deepmodeling/deepmd-kit/assets/51835970/26d57684-1ef5-4fc3-ab5d-9c0368820373)

**Comment by jameswind** [comment]
By looking at the figures it seems that some of the predictions are ok but
some are bad. Can you see where it performs worse? (Say, by looking at the
performance at different temperatures)
Also, I noticed that you used cp2k AIMD. Are you sure that you are printing
out the forces without thermal noises? Please confirm.
If everything above goes reasonably well, then it could be a problem of the
model and there could be several ways to improve the model performance. But
before that let's see more details of data.
You may take a look at this article for your reference:
https://www.pnas.org/doi/full/10.1073/pnas.2302468120?doi=10.1073%2Fpnas.2302468120

**Reply by beidouamg** [reply]
Thanks for your reply. 
In my training set, i include structures from 230 K, 330 K, 430 K to 530 K. When i use the trained potential to check its performance, it has the same problem, mentioned above, at all these temperatures.
As for 'forces without thermal noises', i use the forces calculated from AIMD directly for training, and didn't reevaluate force using single-point calculation.

**Reply by beidouamg** [reply]
@jameswind hello, I am still wondering how I can improve the model performance if the problem is not due to the quality of dataset. Your suggestions are appreciated.

**Comment by njzjz** [comment]
#2704 may help

**Reply by beidouamg** [reply]
Thanks for help.
I think it is not due to the quality of DFT calculations, i checked its convergence and it is converged.
